### PR TITLE
BAH-1846 | Ability to view report images on screen without changing tab

### PR DESCRIPTION
**Pull request by Arjun-Go**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more redable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [x] I have updated the documentation on [Bahmni Wiki](https://bahmni.atlassian.net/wiki/spaces/BAH/overview) or [README](https://github.com/Bahmni/bahmni-lab-frontend/blob/main/README.md) (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
Ability to view report images as pop up on the same screen.

## Screenshots
image

## JIRA tickets
https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/35?modal=detail&selectedIssue=BAH-1846

### CODE REVIEW

File: src/images/image-component.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, {useState} from 'react'
+import {Modal, Button} from 'carbon-components-react'
+import classes from './images-component.scss'
+
+const Images = ({url, fileName}) => {
```
Comment (N0-man) [inline src/images/image-component.tsx]: Images and Image-component does not represent proper component or feature naming... I believe we would want to support `document-preview` as a feature and within it you would have an `image-preview-component` or `pdf-preview-componet`?

File: src/report-table/report-table.component.tsx [hunk 2]
```diff
@@ -40,6 +41,15 @@ const ReportTable = ({patientUuid}) => {
...
       })
       .map(row => {
+        let fileName = ''
+        const contentType = row.resource.presentedForm[0].contentType
+        if (contentType?.endsWith('pdf'))
+          fileName = `${row.resource.presentedForm[0].title}.pdf`
+        else if (contentType?.endsWith('jpg'))
+          fileName = `${row.resource.presentedForm[0].title}.jpg`
+        else if (contentType?.endsWith('jpeg'))
+          fileName = `${row.resource.presentedForm[0].title}.jpeg`
```
Comment (N0-man) [inline src/report-table/report-table.component.tsx]: SRP violation

File: src/report-table/report-table.component.tsx [hunk 3]
```diff
@@ -52,7 +62,7 @@ const ReportTable = ({patientUuid}) => {
             day: '2-digit',
           }),
           requester: '-',
-          file: `${row.resource.presentedForm[0].title}.pdf`,
+          file: fileName,
```
Comment (N0-man) [inline src/report-table/report-table.component.tsx]: you can extract the responsibility to derive filename into its own isolation

```
function deriveFileName(title, contentType) {
	if (contentType.endsWith('pdf'))
		return `${title}.pdf`
  else if (contentType.endsWith('jpg'))
    return `${title}.jpg`
  else if (contentType.endsWith('jpeg'))
    return `${title}.jpeg`
  return ''
}

```
then you could just attach it to the object without the need of any garbage variables or mutations
```
const title = row.resource.presentedForm[0].title
const contentType = row.resource.presentedForm[0].contentType
...
...
          requester: '-',
          file: deriveFileName(title, contentType),
```

File: src/images/image-component.tsx (hunk 1 above)
Comment (Arjun-Go) [reply src/images/image-component.tsx]: We have renamed it to `image-preview-component`

File: src/report-table/report-table.component.tsx (hunk 2 above)
Comment (Arjun-Go) [reply src/report-table/report-table.component.tsx]: Has been refactored

File: src/report-table/report-table.component.tsx (hunk 3 above)
Comment (Arjun-Go) [reply src/report-table/report-table.component.tsx]: Changes have been made according to the inputs above.

**Pull request merged**